cmd/subnetcmd/upgradecmd/generate_upgrade.go

```diff
@@ -0,0 +1,379 @@
+				},
+				"Add an address to amount pair",
+				"Address-Amount",
+				"Ethereum address in Hex format and it's initial amount value, for example: 0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC (address) and 1000000000000000000 (value)",
+			)
+			if err != nil {
+				return err
+			}
+			if cancel {
+				return errors.New("aborted by user")
+			}
+		}
+	}
+	return nil
+}
+
+func (p *nativeMint) ToMap() map[string]interface{} {
+	finalMap := toMap(&p.enabledAddresses, &p.adminAddresses)
+	finalMap[initialMintKey] = p.initialMint
+	return finalMap
+}
+
+func (p *feeManager) PromptParams() error {
+	if err := captureAddress(adminLabel, &p.adminAddresses); err != nil {
+		return err
+	}
+
+	if err := captureAddress(enabledLabel, &p.enabledAddresses); err != nil {
+		return err
+	}
+
+	yes, err := app.Prompt.CaptureYesNo(fmt.Sprintf("Add an '%s' section?", feeConfigKey))
+	if err != nil {
+		return err
+	}
+
+	if yes {
+		chainConfig, _, err := vm.GetFeeConfig(params.ChainConfig{}, app)
+		if err != nil {
+			return err
+		}
+		p.initialFeeConfig = chainConfig.FeeConfig
+	}
+	return nil
+}
+
+func (p *feeManager) ToMap() map[string]interface{} {
+	finalMap := toMap(&p.enabledAddresses, &p.adminAddresses)
+	finalMap[feeConfigKey] = p.initialFeeConfig
+	return finalMap
+}
+
+func (p *contractAllowList) PromptParams() error {
+	return enabledAdminPromptParams(&p.enabledAddresses, &p.adminAddresses)
+}
+
+func (p *contractAllowList) ToMap() map[string]interface{} {
+	return toMap(&p.enabledAddresses, &p.adminAddresses)
+}
+
+func (p *txAllowList) PromptParams() error {
+	return enabledAdminPromptParams(&p.enabledAddresses, &p.adminAddresses)
+}
+
+func (p *txAllowList) ToMap() map[string]interface{} {
+	return toMap(&p.enabledAddresses, &p.adminAddresses)
+}
+
+func enabledAdminPromptParams(enabled *[]common.Address, admin *[]common.Address) error {
+	for {
+		if err := captureAddress(enabledLabel, enabled); err != nil {
+			return err
+		}
+		if err := captureAddress(adminLabel, admin); err != nil {
+			return err
+		}
+
+		if len(*enabled) == 0 && len(*admin) == 0 {
+			ux.Logger.PrintToUser(fmt.Sprintf("We need at least one Ethereum address for either '%s' or '%s'. Otherwise abort.", enabledAddressesKey, adminAddressesKey))
+			continue
+		}
+		return nil
+	}
+}
+
+func toMap(enabledAddresses *[]common.Address, adminAddresses *[]common.Address) map[string]interface{} {
+	finalMap := map[string]interface{}{}
+	if len(*enabledAddresses) > 0 {
+		enabled := make([]string, len(*enabledAddresses))
+		for i := 0; i < len(*enabledAddresses); i++ {
+			enabled[i] = (*enabledAddresses)[i].Hex()
+		}
+		finalMap[enabledAddressesKey] = enabled
+	}
+
+	if len(*adminAddresses) > 0 {
+		admin := make([]string, len(*adminAddresses))
+		for i := 0; i < len(*adminAddresses); i++ {
+			admin[i] = (*adminAddresses)[i].Hex()
+		}
+		finalMap[adminAddressesKey] = admin
+	}
+
+	return finalMap
+}
+
+func captureAddress(which string, addrsField *[]common.Address) error {
```

Comment:
Do we want this to more closely match the interface we use during subnet creation?

Reply:
Maybe it should, but it's most probably going to break some existing code, as the creation process lacks `enabledAddresses` for example.

Maybe better to do in follow up PR via this ticket: https://github.com/ava-labs/avalanche-cli/issues/370